.github/workflows/pr_lint_and_test.yml

```diff
@@ -12,63 +13,74 @@ jobs:
         version: [12, 14, 16]
       fail-fast: false
     steps:
-      - uses: actions/checkout@v3
-      - name: "Use NodeJS"
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Setup NodeJS
         uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.version }}
-      - name: Install npm@8.x
+          cache: "npm"
+      - name: Setup npm
         run: npm i -g npm@next-8
-      - name: "Setup npm"
-        run: |
-          npm set "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}"
-      - name: Install monorepo packages
-        # This installs all the dependencies of ./packages/*
+      - name: Cache node modules
+        id: cache-node-modules
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: "./node_modules"
+          # Use the combo between node version, name, and SHA-256 hash of the lock file as cache key so that
+          # if one of them changes the cache is invalidated/discarded
+          key: ${{ matrix.version }}-build-${{ env.cache-name }}-${{ hashFiles('./package-lock.json') }}
+      - name: Install dependencies
+        # We can skip the install if there was a cache hit
+        if: steps.cache-node-modules.outputs.cache-hit != 'true'
```

Comment:
❤️ this.

Reply:
Question:  I understand that `env.cache-name` will always be the same right?  Any advantage of introducing this env var (and put it in the key)?

Reply:
You are right, it won't change ever. 

I have lifted this from a documentation example and they were using it but now that you mention it I agree it's redundant.

Happy to make another commit and remove it